src/hooks.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2020, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useCallback, useContext, useEffect, useState } from 'react';
+import * as optimizely from '@optimizely/optimizely-sdk';
+import { getLogger } from '@optimizely/js-sdk-logging';
+
+import { setupAutoUpdateListeners } from './autoUpdate';
+import { VariableValuesObject, OnReadyResult } from './client'
+import { OptimizelyContext } from './Context';
+
+const useFeatureLogger = getLogger('useFeature');
+
+type UseFeatureState = {
+  isEnabled: boolean,
+  variables: VariableValuesObject,
+};
+
+type ClientReady = boolean;
+type DidTimeout = boolean;
+
+type UseFeatureOptions = {
+  autoUpdate?: boolean,
+  timeout?: number
+};
+
+type UseFeatureOverrides = {
+  overrideUserId?: string,
+  overrideAttributes?: optimizely.UserAttributes,
+}
+
+interface UseFeature {
+  (
+    featureKey: string,
+    options?: UseFeatureOptions,
+    overrides?: UseFeatureOverrides,
+  ): [
+    UseFeatureState["isEnabled"],
```

Comment:
Does typescript require this to be bracket notation?

Reply:
Yea, since it's not a namespace, you must access the property of a type using bracket notation.

Reply:
Good to know thanks!